### PR TITLE
Allow rectangular shaped images

### PR DESCRIPTION
**Pull request by cdeepakroy**
This PR
* makes all changes necessary to support rectangular shaped images (e.g. resize_val is now a tuple)
* adds a resnet50 based model that is trained in a self-supervised fashion using a siamese triplet loss to preserve temporal coherence on images from minimally invasive surgery videos.

### CODE REVIEW

File: python/smqtk/algorithms/descriptor_generator/pytorch_descriptor.py [hunk 1]
```diff
@@ -190,8 +194,7 @@ def _setup_network(self):
 
         if self.model_uri is not None:
             self._log.debug("load the trained model: {}".format(self.model_uri))
-            snapshot = torch.load(self.model_uri)
-            self.model_cls.load_state_dict(snapshot['state_dict'])
```
Comment (dongshuhao) [inline python/smqtk/algorithms/descriptor_generator/pytorch_descriptor.py]: @cdeepakroy: It is better to save the state_dict instead of save the whole model. I would like to keep the original way to load the model. One example of saving a model is:

torch.save(checkpoint(model, epoch), save_path)

def checkpoint(model, epoch=None):
    package = {
        'epoch': epoch if epoch else 'N/A',
        'state_dict': model.state_dict(),
    }
    return package

File: python/smqtk/pytorch_model/imageNet_ResNet_50.py [hunk 2]
```diff
@@ -45,8 +45,13 @@ def forward(self, x):
 
         return (features, out)
 
+    def load(self, model_file):
```
Comment (dongshuhao) [inline python/smqtk/pytorch_model/imageNet_ResNet_50.py]: @cdeepakroy what's the load function for?

File: python/smqtk/algorithms/descriptor_generator/pytorch_distance_saliency_descriptor.py [hunk 3]
```diff
@@ -437,8 +437,7 @@ def _setup_network(self):
 
         if self.model_uri is not None:
             self._log.debug("load the trained model: {}".format(self.model_uri))
-            snapshot = torch.load(self.model_uri)
-            self.model_cls.load_state_dict(snapshot['state_dict'])
+            self.model_cls.load(self.model_uri)
```
Comment (dongshuhao) [inline python/smqtk/algorithms/descriptor_generator/pytorch_distance_saliency_descriptor.py]: @cdeepakroy : the same here. I would like to keep the original way to load the model
